internal/resources/fleetpolicy.go

```diff
@@ -0,0 +1,458 @@
+	if dsName := packagePolicy.DataStreamName; dsName != "" {
+		return nil, fmt.Errorf("no data stream expected for input package policy %q, found %q", packagePolicy.Name, dsName)
+	}
+
+	policyTemplateName := packagePolicy.TemplateName
+	if policyTemplateName == "" {
+		name, err := findPolicyTemplateForInputPackage(manifest, packagePolicy.InputName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to determine the associated policy_template: %w", err)
+		}
+		policyTemplateName = name
+	}
+
+	policyTemplate, err := selectPolicyTemplateByName(manifest.PolicyTemplates, policyTemplateName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find the selected policy_template: %w", err)
+	}
+
+	ds := kibana.PackageDataStream{
+		Name:      packagePolicy.Name,
+		Namespace: policy.Namespace,
+		PolicyID:  policy.ID,
+		Enabled:   !packagePolicy.Disabled,
+		Inputs: []kibana.Input{
+			{
+				PolicyTemplate: policyTemplate.Name,
+				Enabled:        !packagePolicy.Disabled,
+			},
+		},
+	}
+	ds.Package.Name = manifest.Name
+	ds.Package.Title = manifest.Title
+	ds.Package.Version = manifest.Version
+
+	streamInput := policyTemplate.Input
+	ds.Inputs[0].Type = streamInput
+
+	dataset := fmt.Sprintf("%s.%s", manifest.Name, policyTemplate.Name)
+	streams := []kibana.Stream{
+		{
+			ID:      fmt.Sprintf("%s-%s.%s", streamInput, manifest.Name, policyTemplate.Name),
+			Enabled: !packagePolicy.Disabled,
+			DataStream: kibana.DataStream{
+				Type:    policyTemplate.Type,
+				Dataset: dataset,
+			},
+		},
+	}
+
+	// Add policyTemplate-level vars.
+	vars := setKibanaVariables(policyTemplate.Vars, common.MapStr(packagePolicy.Vars))
+	if _, found := vars["data_stream.dataset"]; !found {
+		var value packages.VarValue
+		value.Unpack(dataset)
+		vars["data_stream.dataset"] = kibana.Var{
+			Value: value,
+			Type:  "text",
+		}
+	}
+
+	streams[0].Vars = vars
+	ds.Inputs[0].Streams = streams
+
+	return &ds, nil
+}
+
+func setKibanaVariables(definitions []packages.Variable, values common.MapStr) kibana.Vars {
+	vars := kibana.Vars{}
+	for _, definition := range definitions {
+		val := definition.Default
+
+		value, err := values.GetValue(definition.Name)
+		if err == nil {
+			val = packages.VarValue{}
+			val.Unpack(value)
+		}
+
+		vars[definition.Name] = kibana.Var{
+			Type:  definition.Type,
+			Value: val,
+		}
+	}
+	return vars
+}
+
+func (f *FleetAgentPolicy) Update(ctx resource.Context) error {
+	if f.Absent {
+		provider, err := f.provider(ctx)
+		if err != nil {
+			return err
+		}
+
+		err = provider.Client.DeletePolicy(ctx, f.ID)
+		if err != nil {
+			return fmt.Errorf("could not delete policy %q: %w", f.Name, err)
+		}
+
+		return nil
+	}
+
+	return errors.New("update not implemented")
+}
+
+type FleetAgentPolicyState struct {
+	current  *kibana.Policy
+	expected bool
+}
+
+func (s *FleetAgentPolicyState) Found() bool {
+	return !s.expected || s.current != nil
```

Comment:
> Is this to ensure that the deletion process is triggered always ?

The decision will be made by `NeedsUpdate` in this case. If we return true here, the manager will call `Create`, what we don't want if the resource is absent.

The truth is that all this logic is a bit convoluted for resources with `absent: true`. I did it this way considering that maybe not all resources are going to have something like `absent`, but so far we have it everywhere, and complicates the logic to check state and to update.

If we consider that all kinds of resources can be `absent`, we could have a CRUD interface, instead of the current "CRU" and a more straight-forward logic in these cases.

Reply:
Follow up issue created about this https://github.com/elastic/go-resource/issues/11.